### PR TITLE
Change bulk operation for metrics and index templates to support Open…

### PR DESCRIPTION
**Pull request by travisbenedict**
…Search 2.0 as a data store

Signed-off-by: travisbenedict <travisrb10@gmail.com>

### Description
Add support for OpenSearch 2.0 as a metrics data store by updating the MDS index templates, removing the `doc_type` argument from bulk requests and removing `include_type_name` from put template requests. 

### Issues Resolved
Closes https://github.com/opensearch-project/opensearch-benchmark/issues/150
 
### Check List
- [ ] New functionality includes testing
  - [ ] All unit and integration tests pass
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by engechas** [comment]
This change was discussed a bit in #108. By merging this PR we are dropping support for metrics stores <ES7. This seems fine to me but there are sure to be other cases of ES/OS conflict in the future. It would be good to align on a decision about how to handle any and all of these conflicts going forwards.

**Comment by travisbenedict** [comment]
> This change was discussed a bit in #108. By merging this PR we are dropping support for metrics stores <ES7. This seems fine to me but there are sure to be other cases of ES/OS conflict in the future. It would be good to align on a decision about how to handle any and all of these conflicts going forwards.

I completely agree with you. A version support policy is something I'm planning to publish soon.

In this particular case I would argue that we implicitly dropped support for Elasticsearch as a data store back in https://github.com/opensearch-project/opensearch-benchmark/pull/74 prior to the alpha release

File: osbenchmark/metrics.py [hunk 1]
```diff
@@ -102,13 +93,7 @@ def refresh(self, index):
         return self.guarded(self._client.indices.refresh, index=index)
 
     def bulk_index(self, index, doc_type, items):
-        # TODO #653: Remove version-specific support for metrics stores before 7.0.0.
-        # pylint: disable=import-outside-toplevel
-        import opensearchpy.helpers
-        if self._cluster_version[0] > 6:
-            self.guarded(opensearchpy.helpers.bulk, self._client, items, index=index, chunk_size=5000)
-        else:
-            self.guarded(opensearchpy.helpers.bulk, self._client, items, index=index, doc_type=doc_type, chunk_size=5000)
```
Comment (achitojha) [inline osbenchmark/metrics.py:111]: Can you add some context on why removing metric store support for ES < 6  is needed here?

Comment (travisbenedict) [reply osbenchmark/metrics.py:111]: Removing support for ES 6 and below is not explicitly needed but would require an additional maintenance burden for us. ES 7 deprecated certain parameters related to put template operations (more info [here](https://www.elastic.co/guide/en/elasticsearch/reference/7.17/removal-of-types.html)). If we want to continue supporting ES 6 as a data store we would need to create duplicates of the index templates and add some additional branching logic.

**Review by engechas — APPROVED** [review]

**Review by IanHoang — APPROVED** [review]

**Pull request merged**